src/main/java/org/breedinginsight/services/TraitService.java

```diff
@@ -59,4 +83,107 @@
 
        return traitDAO.getTraitFull(programId, traitId);
     }
+
+    public List<Trait> createTraits(UUID programId, List<Trait> traits, AuthenticatedUser actingUser)
+            throws DoesNotExistException, ValidatorException {
+
+        Optional<Program> optionalProgram = programService.getById(programId);
+        if (!optionalProgram.isPresent()) {
```

Comment:
you could get fancy on these optional checks if you want: 
```Program program = optionalProgram.orElseThrow(() -> new  DoesNotExistException("Program does not exist"));```

Reply:
You keep your voodoo magic out of my pull request!!!

Nice, probably won't make changes in this PR, but I'll start using that in the future!